packages/expo-modules-core/ios/Core/Views/AppleView.swift

```diff
@@ -0,0 +1,48 @@
+// Copyright 2015-present 650 Industries. All rights reserved.
+
+import SwiftUI
+
+/**
+ An abstract view supports both UIKit and SwiftUI.
+ */
+public enum AppleView {
+  case uikit(UIView)
+  case swiftui(any ExpoSwiftUI.View)
+
+  static func from(_ view: UIView?) -> AppleView? {
+    guard let view else {
+      return nil
+    }
+    return .uikit(view)
+  }
+
+  static func from(_ view: (any ExpoSwiftUI.View)?) -> AppleView? {
+    guard let view else {
+      return nil
+    }
+    return .swiftui(view)
+  }
+
+  func toUIView() throws -> UIView {
+    switch self {
+    case .uikit(let view):
+      return view
+    case .swiftui:
+      throw AppleViewConversionError.cannotConvertSwiftUIViewToUIView
+    }
+  }
+
+  func toSwiftUIView() throws -> any ExpoSwiftUI.View {
+    switch self {
+    case .uikit:
+      throw AppleViewConversionError.cannotConvertUIKitViewToSwiftUIView
+    case .swiftui(let swiftUIView):
+      return swiftUIView
+    }
+  }
+}
```

Comment:
```suggestion
public enum AppleView {
  case uikit(UIView)
  case swiftui(any ExpoSwiftUI.View)

  static func from<T>(_ view: T?) -> AppleView? {
    guard let view else {
      return nil
    }
    if let uiView = view as? UIView {
      return .uikit(uiView)
    } else if let swiftUIView = view as? any ExpoSwiftUI.View {
      return .swiftui(swiftUIView)
    }
    return nil
  }
  
  private func toView<T>() throws -> T {
    switch self {
    case .uikit(let view):
      if let result = view as? T {
        return result
      }
      throw AppleViewConversionError.cannotConvertUIKitViewToSwiftUIView
    case .swiftui(let view):
      if let result = view as? T {
        return result
      }
      throw AppleViewConversionError.cannotConvertSwiftUIViewToUIView
    }
  }

  func toUIView() throws -> UIView {
    try toView()
  }

  func toSwiftUIView() throws -> any ExpoSwiftUI.View {
    try toView() as any ExpoSwiftUI.View
  }
}
```
Could we merge some of these functions and behaviour?

Reply:
to have
- `static func from(_ view: UIView?) -> AppleView?`
- `static func from(_ view: (any ExpoSwiftUI.View)?) -> AppleView?`

than the `static func from(_ view: UIView?) -> AppleView?`, the benefit is that we can leverage compiler for type checking. if people pass input other than UIView and ExpoSwiftUI.View, compiler would help to catch the unexpected input.


then the second `private func toView<T>()` doesn't save much in my opinion. the original `toUIView` / `toSwiftUIView` looks more straightforward